source/server/listener_impl.cc

```diff
@@ -309,7 +326,6 @@ ListenerImpl::~ListenerImpl() {
   // a local init manager we should block the notification from trying to move us from warming to
```

Comment:
Ok, I think I follow what happened here in terms of changing having different init managers and resetting this watch, vs having the new target and arrangement of firings.

However, there are some outdated comments throughout this file. The above for instance, as this destructor does not any longer reset the init_watcher_. Do you mind going through the file and deleting and updating comments. I think the process here is tricky and being extra crisp on what is going on would make the next person reading this have an easier time.